rdt/transformers/text.py

```diff
@@ -156,47 +188,44 @@ def _reverse_transform(self, data):
         else:
             sample_size = self.data_length
 
-        if sample_size > self.generator_size:
-            if self.enforce_uniqueness:
-                raise TransformerProcessingError(
-                    f'The regex is not able to generate {sample_size} unique values. '
-                    f"Please use a different regex for column ('{self.get_input_column()}')."
-                )
-
-            LOGGER.info(
-                "The data has %s rows but the regex for '%s' can only create %s unique values."
-                " Some values in '%s' may be repeated.",
-                sample_size, self.get_input_column(), self.generator_size, self.get_input_column()
-            )
+        self._warn_not_enough_unique_values(sample_size)
 
         remaining = self.generator_size - self.generated
-        if sample_size > self.generator_size - self.generated:
-            if self.enforce_uniqueness:
-                raise TransformerProcessingError(
-                    f'The regex generator is not able to generate {sample_size} new unique '
-                    f'values (only {remaining} unique value left). Please use '
-                    "'reset_randomization' in order to restart the generator."
-                )
-
+        if sample_size > remaining:
             self.reset_randomization()
             remaining = self.generator_size
 
         if remaining >= sample_size:
-            reverse_transformed = np.array([
-                next(self.generator)
-                for _ in range(sample_size)
-            ], dtype=object)
-
+            reverse_transformed = [next(self.generator) for _ in range(sample_size)]
             self.generated += sample_size
 
         else:
-            self.generated = self.generator_size
             generated_values = list(self.generator)
-            reverse_transformed = []
-            while len(reverse_transformed) < sample_size:
-                remaining_samples = sample_size - len(reverse_transformed)
-                reverse_transformed.extend(generated_values[:remaining_samples])
-
-            reverse_transformed = np.array(reverse_transformed, dtype=object)
-
-        return reverse_transformed
+            reverse_transformed = generated_values[:]
+            self.generated = self.generator_size
+            if self.enforce_uniqueness:
+                try:
+                    int(generated_values[0])
+                    dtype = int
+                    counter = int(generated_values[-1]) + 1
+                except ValueError:
+                    dtype = 'other'
+                    counter = 0
+
+                while len(reverse_transformed) < sample_size:
+                    remaining_samples = sample_size - len(reverse_transformed)
+                    if dtype == int:
```

Comment:
Do we need the `while` boucle for the `int` case? It seems likely to resolve in one iteration

Reply:
Thanks, I missed that.